enhancements/vertical-pod-autoscaler/add-vpa-to-operator-hub.md

```diff
@@ -0,0 +1,191 @@
+---
+title: vpa-available-on-operatorhub
+authors:
+  - "@joelsmith"
+reviewers:
+  - "@rphillips"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-10-08
+last-updated: 2019-10-10
+status: provisional
+---
+
+# Make Vertical Pod Autoscaler available on OperatorHub
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. Which global VPA controller options should we expose? The proposed set of options are as follows:
+
+ 	- From the [admission-controller](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/main.go#L47-L58): none
+	- From the [updater](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/updater/main.go#L39-L48)  (see also [here](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go#L40-L42)): none
```

Comment:
can you clarify what min replicas = 2 means?

Reply:
For any group of pods having the same creator (e.g. deployment, replicaset, etc), there must be at least `min-replicas` pods running before the VPA updater will take action. So if the pod is a singleton and the default `min-replicas` value of 2 is used, the VPA will never kill the pod automatically, and so the new limits will only take effect if the pod is manually restarted (or restarted by some other process). Unfortunately, this is a global limit (as opposed to a global default limit that could be overridden by individual VPA objects). I think it may have been preferable to allow each VPA object to specify this limit so that the acceptable level of disruption could be specified on a case-by-case basis. That might be something worth proposing as a future enhancement.